src/CommandingOfficers/Tech.java

```diff
@@ -0,0 +1,514 @@
+        u.alterHP(healAmount, true);
+      }
+    }
+  }
+
+  /** Drop a MechWarrior onto the most contested point on the battlefront. */
+  private static class TechdropAbility extends CommanderAbility
+  {
+    private static final long serialVersionUID = 1L;
+
+    private CODamageModifier damageBuff = null;
+    private CODefenseModifier defenseBuff = null;
+    private int dropRange;
+    private int numDrops;
+    private UnitModel unitModelToDrop;
+
+    GameEventQueue abilityEvents;
+
+    TechdropAbility(Commander commander, String abilityName, double abilityCost, UnitModel unitToDrop, int buff, int num, int abilityRange)
+    {
+      super(commander, abilityName, abilityCost);
+
+      unitModelToDrop = unitToDrop;
+
+      // Create COModifiers that we can apply when needed.
+      damageBuff = new CODamageModifier(buff);
+      defenseBuff = new CODefenseModifier(buff);
+      numDrops = num;
+      dropRange = abilityRange;
+      abilityEvents = new GameEventQueue();
+
+      // Only mechanical units get the firepower boost.
+      long exclude = UnitModel.TROOP;
+      ArrayList<UnitModel> models = commander.getAllModels(Tech.mechanicalUnits, true, exclude);
+      for( UnitModel m : models )
+        damageBuff.addApplicableUnitModel(m);
+    }
+
+    @Override
+    protected void perform(MapMaster gameMap)
+    {
+      // Bump up our power level.
+      myCommander.addCOModifier(damageBuff);
+      myCommander.addCOModifier(defenseBuff);
+
+      // Perform any events we generated in getEvents().
+      while(!abilityEvents.isEmpty())
+      {
+        abilityEvents.poll().performEvent(gameMap);
+      }
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      boolean log = true;
+      if( log ) System.out.println("[TechDrop] getEvents() entry");
+
+      // Prep events for `numDrops` deployments. Recalculate landing position between each.
+      Set<XYCoord> dropLocs = new HashSet<XYCoord>();
+      for( int i = 0; i < numDrops; ++i )
+      {
+        TeleportEvent techDrop = generateDropEvent(gameMap, dropLocs, log);
+        if( null != techDrop )
+        {
+          // Store the new event.
+          dropLocs.add(techDrop.getEndPoint());
+          abilityEvents.add(techDrop);
+        }
+        else
+        {
+          System.out.println("[TechdropAbility.getEvents] WARNING! Could not find drop location.");
+        }
+      }
+
+      return abilityEvents;
+    }
+
+    private TeleportEvent generateDropEvent(MapMaster gameMap, Set<XYCoord> priorDrops, boolean log)
+    {
+      // Create a new Unit to drop onto the battlefield.
+      Unit techMech = new Unit(myCommander, unitModelToDrop);
+      techMech.isTurnOver = false; // Hit the ground ready to rumble.
+      if( myCommander.HQLocation.xCoord >= gameMap.mapWidth / 2 )
+        techMech.x = gameMap.mapWidth; // Make the unit face left as it falls to match the rest of the units.
```

Comment:
...ah, I see you've accounted for my other comment here.

Reply:
Only works if you drop on the correct side of the HQ. More responses on the other comment.